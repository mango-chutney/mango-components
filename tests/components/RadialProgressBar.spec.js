// @flow

import 'raf/polyfill';
import 'jest-styled-components';
import React from 'react';
import renderer from 'react-test-renderer';
import MC from '../../src';

const RadialProgressBar = MC.RadialProgressBar.createComponent();

function RadialProgressBarComponent() {
  return (
    <RadialProgressBar
      backgroundColor="#ffca00"
      percentage={75}
      width={100}
      strokeWidthScale={20}
      strokeLinecap="round"
    />
  );
}

it('RadialProgressBar component renders correctly', () => {
  const tree = renderer.create(<RadialProgressBarComponent />).toJSON();
  expect(tree).toMatchSnapshot();
});
