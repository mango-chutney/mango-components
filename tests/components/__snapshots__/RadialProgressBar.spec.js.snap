// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadialProgressBar component renders correctly 1`] = `
.c0 {
  height: 6.25rem;
  position: relative;
  width: 6.25rem;
}

.c1 {
  height: 100%;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-transform: rotateZ(-90deg);
  -ms-transform: rotateZ(-90deg);
  transform: rotateZ(-90deg);
  width: 100%;
}

.c1 svg {
  overflow: visible;
}

.c1 circle {
  -webkit-transition: stroke-dashoffset 0.5s linear;
  transition: stroke-dashoffset 0.5s linear;
}

.c2 {
  stroke: #e2e7ee;
}

.c3 {
  stroke: #ffca00;
  -webkit-animation: iWFGQb 0.5s normal forwards;
  animation: iWFGQb 0.5s normal forwards;
}

<div
  className="c0"
  width={100}
>
  <div
    className="c1"
  >
    <svg
      height="100%"
      viewBox="0 0 100 100"
      width="100%"
    >
      <circle
        className="c2"
        cx={50}
        cy={50}
        fill="none"
        r={40}
        strokeWidth={11.11111111111111}
      />
      <circle
        className="c3"
        cx={50}
        cy={50}
        fill="none"
        r={40}
        strokeDasharray={251}
        strokeDashoffset={62.75}
        strokeLinecap="round"
        strokeWidth={11.11111111111111}
      />
    </svg>
  </div>
</div>
`;
